python/tank_vendor/shotgun_authentication/session_cache.py

```
@@ -0,0 +1,378 @@
+    )
+
+
+def _get_global_authentication_file_location():
+    """
+    Returns the location of the authentication file on disk. This file
+    stores authentication related information for all sites. At this moment,
+    the file stores only the current host.
+
+    :returns: Path to the login information.
+    """
+    return os.path.join(
+        _get_cache_location(),
+        "authentication.yml"
+    )
+
+
+def _get_site_authentication_file_location(base_url):
+    """
+    Returns the location of the users file on disk for a specific site.
+
+    :param base_url: The site we want the login information for.
+
+    :returns: Path to the login information.
+    """
+    return os.path.join(
+        _get_site_cache_location(base_url),
+        "authentication.yml"
+    )
+
+
+def _ensure_folder_for_file(filepath):
+    """
+    Makes sure the folder exists for a given file.
+
+    :param filepath: Path to the file we want to make sure the parent directory
+                     exists.
+
+    :returns: The path to the file.
+    """
+
+    folder, _ = os.path.split(filepath)
+    if not os.path.exists(folder):
+        old_umask = os.umask(0077)
+        try:
+            os.makedirs(folder, 0700)
+        finally:
+            os.umask(old_umask)
+    return filepath
+
+
+def _try_load_yaml_file(file_path):
+    """
+    Loads a yaml file.
+
+    :param file_path: The yaml file to load.
+
+    :returns: The dictionary for this yaml file. If the file doesn't exist or is
+              corrupted, returns an empty dictionary.
+    """
+
+    if not os.path.exists(file_path):
+        logger.debug("Yaml file missing: %s" % file_path)
+        return {}
+    try:
+        # Open the file and read it.
+        with open(file_path, "r") as config_file:
+            return yaml.load(config_file)
+    except yaml.YAMLError:
+        # Return to the beginning
+        config_file.seek(0)
+        logger.error("File '%s' is corrupted!" % file_path)
```

I agree. Let's swallow the exception, return {} and call logger.exception.